### PR TITLE
Use membership common 179

### PR DESCRIPTION
**Pull request by tomverran**
This preserves subscription type information and removes some very horrible unsafe casts
I am doing this so I can split up the upgrade controller method to return JSON for paid to paid upgrades

### CODE REVIEW

File: frontend/app/actions/ActionRefiners.scala [hunk 1]
```diff
@@ -49,15 +49,17 @@ object ActionRefiners extends LazyLogging {
   type SubReqWithSub[A] = SubscriptionRequest[A] with Subscriber
 
   private def getSubRequest[A](request: AuthRequest[A]): Future[Option[SubReqWithSub[A]]] = {
+
```
Comment (AWare) [inline frontend/app/actions/ActionRefiners.scala]: new whitespace?

File: frontend/app/actions/ActionRefiners.scala [hunk 2]
```diff
@@ -68,26 +70,21 @@ object ActionRefiners extends LazyLogging {
...
-      Future.successful(request.subscriber match {
-        case FreeMember(mem) => Right(new SubscriptionRequest[A](request) with FreeSubscriber {
-          override def subscriber = mem
-        })
-        case _ => Left(onPaidMember(request))
-      })
+      Future.successful(request.paidOrFreeSubscriber.bimap(free =>
+        new SubscriptionRequest[A](request) with FreeSubscriber {
+          override def subscriber = free
+        }, _ => onPaidMember(request)).swap.toEither
```
Comment (AWare) [inline frontend/app/actions/ActionRefiners.scala]: Maybe a comment would help?

Comment (tomverran) [inline frontend/app/actions/ActionRefiners.scala]: I feel that I'd end up writing something like
```scala
[...].swap.toEither //swaps the disjunction then converts it to an either
```
Which feels a bit redundant

**Comment by tomverran** [comment]
So I'm going to merge this on Tuesday

**Comment by AWare** [comment]
:+1:  :)

**Comment by tomverran** [comment]
ran through the join / upgrade flows and all seemed fine

**Comment by tomverran** [comment]
(this is live)

**Pull request merged**